core/src/main/java/org/elasticsearch/discovery/zen/publish/PendingClusterStatesQueue.java

```diff
@@ -164,12 +164,17 @@ public synchronized void markAsProcessed(ClusterState state) {
                             currentMaster
                     );
                 }
-            } else if (state.supersedes(pendingState) && pendingContext.committed()) {
+            } else if (state.version() >= pendingState.version()) {
+                assert state.supersedes(pendingState) || (
```

Comment:
Removed assertion, reordered checks in c2ed5a1c9e4c7decb73eb026e7f7a8591909389f.